### ISSUE TITLE
add hasComment value in Text value key

### ISSUE THREAD

**Issue by anaskhanstack** [surveyjs/survey-library#2476]
{
             "value": "other",
             "name": "other",
             "text": "other: ![other](" + assetsurl + "Group13452.svg =75x75)",
             "hasComment": true,
             "enableIf": "{QDietPref} notcontains ['None']",
}

want to achieve hasComment value in "value" key

**Comment by andrewtelnov** [comment]
@anaskhanstack You can easility add "hasComment" property.
``Survey.Serializer.addProperty("itemvalue", "hasComment:boolean");``.
However, you will have to write code using [``survey.onAfterRenderQuestion`` event](https://surveyjs.io/Examples/Library?id=survey-afterrender).
The common solution is to have a separate "comment" question with visibleIf property as I answered in [previous ticket](https://github.com/surveyjs/survey-library/issues/2475).

Thank you,
Andrew

**Comment by anaskhanstack** [comment]
Can you please make the sample code in plunker?
Thank you,
Anas Khan

**Comment by andrewtelnov** [comment]
@anaskhanstack It is a standard functionality. Here is [the example](https://surveyjs.io/Examples/Library?id=real-nps) that explains it.

Thank you,
Andrew

**Comment by anaskhanstack** [comment]
I don't want to make another question, so how can i achieve that using single question 
Thank you 
Anas Khan

**Comment by andrewtelnov** [comment]
@anaskhanstack You will have to write a lot of code on survey.onAfterRenderQuestion event or create a [custom widget](https://surveyjs.io/Documentation/Survey-Creator?id=Create-Custom-Widget). To create a plunker example we will need to spend several hours to make it work, unfortunately we do not have this time.

Thank you,
Andrew

**Comment by anaskhanstack** [comment]
"choices": [
                        {
                            "value": "Sesame",
                            "enableIf": "{QAllergy} notcontains ['None']",
                            "text": "Sesame![Sesame](" + assetsurl + "Group13458.svg =75x75)"
                        },
                        {
                            "value": "Peanuts",
                            "text": "Peanuts![Peanuts](" + assetsurl + "peanut.svg =75x75)",
                            "enableIf": "{QAllergy} notcontains ['None']"
                        }
]

So basically i can not make e.g. "value":"peanut" choice type comment?

**Comment by andrewtelnov** [comment]
@anaskhanstack What do you mean? Do you want to show comment when a use chooses the value "Peanuts", am I correct? We do not have this functionality right now. 

Thank you,
Andrew

**Comment by anaskhanstack** [comment]
> @anaskhanstack What do you mean? Do you want to show comment when a use chooses the value "Peanuts", am I correct? We do not have this functionality right now.
> 
> Thank you,
> Andrew

Yes i was asking that question, if the functionality is not implemented so your team should be having any alternative example for it, so that they can use that example.
Thank you 
Anas Khan

**Comment by andrewtelnov** [comment]
@anaskhanstack If there be more requests, we will make it. Right now the most people are happy with "other" and seperate comment question with visibleIf functionality. 

Here is what stop us of implementing it:
1) Almost nobody needs it
2) We will have an additional property that we have to show somewhere
3) We have to store value for this additional comment somewhere. Some people may want to store it separately from origional question. We have to think about it.
4) We have to show this new value in the analytics pack and add some options in that library.
5) Change rendering for radiogroup/drodown/checkbox for all platforms we support
6) Write functional tests for SurveyJS Library and Analytics Pack.
Since it would require at least couple days to implement this feature out of the box and the functionality has a little value for our users/customers, the ratio: value/cost is low and this feature  gos into the end of our TODO list. 
Where "value" is how many of our users/customers need it and how bad they need it, from show stopper, to would be nice to have, and "cost" is developers hours.
If we see many people needs it, we can change our mind, since the ratio will be better. Right now we have a lot of other tasks with way better ratio: value/cost.
This is as simple as this.

PS: We are open source project. You can always fork our repository and add the missing functionality in the way you need it.

Thank you,
Andrew

**Comment by anaskhanstack** [comment]
Ok i understand, but on large scale application there will be such use-cases and you guys need to be ready for that otherwise the developer will get frustrated
Here is why i want that to be implemented as a developer:
1. we can not put enable if on hasOther for disabling the choice.
2. if we put each question in single page which contain more than 8 choices and each choice contains animation of its contents, 
    it will be very hard for the users to add a new question for only hasOther and resolve all the animations and make additional 
    conditions i.e. when it come to x question that contains hasOther specific question so remove all x hasOther animation and 
    add new custom css to be the part of x question so that it give client illusion that this is not a new question.  
3. If we have 38 question and each question covers whole webpage and contain 6 choice + hasOther choice, we will create 38 
    new question for hasOther and at the end we will have to join that to new question(that contain hasOther which are 38) with 
    right question number to show the result at the end on result card  

At the end my feedback is not to show that this library is not good or what so ever, but is to improve the library because honestly i find this library useful, so no hard feeling for library.
Thank you
Anas Khan
